### PR TITLE
Add WriteOptions.no_slowdown

### PR DESCRIPTION
**Pull request by maysamyabandeh**
Summary:
If the WriteOptions.no_slowdown flag is set AND we need to wait or sleep for
the write request, then fail immediately with Status::Incomplete().

Test Plan:
unit test added

### CODE REVIEW

File: db/write_thread.cc [hunk 1]
```diff
@@ -268,6 +268,12 @@ size_t WriteThread::EnterAsBatchGroupLeader(
       break;
     }
 
+    if (!w->no_sleep && leader->no_sleep) {
```
Comment (siying) [inline db/write_thread.cc]: I think we need to break for the other way around too.

In the disableWAL case, a query disabling WAL can go with one without, but it's not the case for no waiting.

**Comment by maysamyabandeh** [comment]
Sure, Thanks. Let me also update no_sleep to no_slowdown as per the original suggestion.

Also can you double check the change here: https://github.com/facebook/rocksdb/pull/1527/files#diff-71a34e2e4e097b44064da7d3a2b68bd8R5025 It will also affect the code path for the requests that fail for reasosn other than slowdown.

**Comment by siying** [comment]
@maysamyabandeh it is true that it may impact failures for other reasons. I can't simply see anything wrong with the diff. If you are not sure, I suggest you add unit tests to verify it.

**Comment by maysamyabandeh** [comment]
The change does makes sense to me too, just wanted to make sure it is viewed by another pair of eyes.

**Comment by siying** [comment]
Great!

**Comment by siying** [comment]
Any news here?

**Comment by maysamyabandeh** [comment]
Some sandcastle tasks timeout with no indicative output. I can go ahead landing it if I get a +1.

**Comment by siying** [comment]
@maysamyabandeh valgrind is failing in master. If it fails with some NewIterator error, ignore it and just land.

**Comment by maysamyabandeh** [comment]
Unfortunately it is more than that:
```
Running	unit_non_shm
Running	unit_481
Running	asan
Running	valgrind
Running	no_compression
Running	tsan
Running	clang_unit
Running	punit
Running	unit
Running	code_cov
```
I already kicked a re-run but they timed out again.

**Comment by maysamyabandeh** [comment]
@siying travis fails at  ./db_dynamic_level_test with "Sst file size mismatch" error which is not likely to be caused by this patch. What do you think? 
https://travis-ci.org/facebook/rocksdb/jobs/177802346#L1379

Also this assert failure at WriteBatchThreadedTest:
https://travis-ci.org/facebook/rocksdb/jobs/177802351#L1024

**Comment by siying** [comment]
Yes they are unrelated. Just land it!

**Pull request closed without merging**